src/kv/test/kv_test.cpp

```diff
@@ -1471,169 +1471,202 @@ TEST_CASE("Conflict resolution")
   REQUIRE_THROWS(tx2.commit());
 }
 
-TEST_CASE("Primary can create correct execution order")
+std::string rand_string(size_t i)
+{
+  return fmt::format("{}: {}", i, rand());
+}
+
+TEST_CASE("Max conflict version tracks execution order")
```

Comment:
Took me a while to debug why this test was broken, and it turned out to be because it was using `id` as a `kv::Version` (return value from `version_resolver`). While fixing that I combined the good and bad tests into 1 to reduce the duplicated code, removed a type confusion between primary and backup, and added 2 entries to the test matrix testing in-primary-order execution (for both independent and strictly-ordered transactions).